src/envoy/http/authn/origin_authenticator.cc

```diff
@@ -57,10 +58,29 @@ bool OriginAuthenticator::run(Payload* payload) {
     }
   }
 
+  bool triggered = false;
+  const char* request_path = nullptr;
+  if (filter_context()->headerMap().Path() != nullptr) {
+    request_path = filter_context()->headerMap().Path()->value().c_str();
+    ENVOY_LOG(debug, "Got request path {}", request_path);
+  } else {
+    ENVOY_LOG(error,
+              "Failed to get request path, JWT will always be used for "
+              "validation");
+  }
+
   for (const auto& method : policy_.origins()) {
-    if (validateJwt(method.jwt(), payload)) {
-      success = true;
-      break;
+    const auto& jwt = method.jwt();
+
+    if (AuthnUtils::IsJwtTriggered(request_path, jwt)) {
```

Comment:
Upstream have per-route or per-path requirement config. https://github.com/envoyproxy/envoy/blob/master/api/envoy/config/filter/http/jwt_authn/v2alpha/config.proto#L401

It is not clear to me how this implementation aligns with it.  If not, I think you have to re-do it with you use Envoy jwt_authn filter.

Reply:
This PR is for istio authn filter and the decision of if using a JWT is made in the authn filter. I think the Envoy jwt_authn is currently not used in proxy, right? Switching to it is a larger refactor that is independent of this PR.

After switched to the Envoy jwt_authn filter, we can update the implementation if we decided it's better to do this in jwt_authn filter instead of our proxy authn filter. There should be no user facing difference in two implementations.

@diemtvu WDYT?

Reply:
My point is:  have you looked at the Envoy jwt_authn per-path requirement config?   How easy is the current authn per-path config align with that config so it is easier to switch in the future? 

I just briefly looked at your config, e.g. inclusive, exclusive stuff.  Envoy config doesn't have that.  It will be hard to switch in the future.

Reply:
I agree with @qiwzhang. It is better to align with upstream JWT filter now. Otherwise, we keep adding more logic in proxy code, which will have to be rewritten when we switch to upstream Envoy filter.

Reply:
@qiwzhang 
The current authn per-path semantic is a subset of the upstream JWT filter.
For a `included_path`, it could be converted to a `JwtRequirement` with `provider_name`, which means a specific path must have the specified JWT validation. 
For a `excluded_path`, it should be converted to a `JetRequirement` without provider, which means the path doesn't require jwt validation, e.g. `/health`.


@liminw The current authn per-path semantic is a subset of the upstream JWT filter. It takes time to switch to upstream Envoy filter and when that is done, we still need to convert the existing pilot code to generate corresponding config. For now, we don't need to modify pilot code and the only change we need is in this PR.

Reply:
I just had a second thought on this, it's okay for single JWT case but it turns out to be a problem when using multiple JWT with `included_paths` and `excluded_paths`. So I'll update the API design to reflect this.

Reply:
I think I mentioned that (upstream jwt) to @yangminzhu at some point, though after some thoughts, I think it's ok as we may not to implement / use this feature in JWT filter. This will be used by authn_filter only, JWT filter still do validation regardless of the path. 

Being said that, I also realized that bundle this information inside JWT params may cause some confusion. On the other hand, to customers (who don't care or know about envoy implementation details), it may be ok. Thoughts?

Reply:
If it's okay to use this in authn_filter only, I think I would like to go with this way. The reason is if we really want to align with the upstream jwt config completely, we have to make some breaking changes (the user have to update their existing manually) to the current  [Policy](https://github.com/istio/api/blob/fac61dc6aaec82b5824918f379e29ff48afcd2cb/authentication/v1alpha1/policy.proto#L311).  And we tried our best to make this change backward compatible in the design doc.

The upstream jwt config focused on path first and then connect the path to a group of named JWTs, however, in proxy authn filter, we focused on JWT first, The JWT don't have name and we don't support any complex `and`/`or` logic.

@liminw  @qiwzhang WDYT?